lib/kafka/client.rb

```diff
@@ -40,8 +40,13 @@ class Client
     # @param ssl_client_cert_key [String, nil] a PEM encoded client cert key to use with an
     #   SSL connection. Must be used in combination with ssl_client_cert.
     #
+    # @param principal [String, nil] a KRB5 principal
+    #
+    # @param keytab [String, nil] a KRB5 keytab filepath
+    #
     # @return [Client]
-    def initialize(seed_brokers:, client_id: "ruby-kafka", logger: nil, connect_timeout: nil, socket_timeout: nil, ssl_ca_cert: nil, ssl_client_cert: nil, ssl_client_cert_key: nil)
+    def initialize(seed_brokers:, client_id: "ruby-kafka", logger: nil, connect_timeout: nil, socket_timeout: nil,
+                   ssl_ca_cert: nil, ssl_client_cert: nil, ssl_client_cert_key: nil, principal: nil, keytab: nil)
```

Comment:
great. Maybe - better to use `gssapi` instead of `krb5`, as it's commonly used in kafka docs and configs. so I'd change `principal` to `sasl_gssapi_principal` and `keytab` to `sasl_gssapi_keytab`